File: src/main/java/com/github/bin/mapper/msg/HisMsgMapper.java
package com.github.bin.mapper.msg;

import com.github.bin.entity.msg.HisMsg;
import org.apache.ibatis.annotations.Param;

import java.util.List;

/**
 * @author bin
 * @since 2023/08/22
 */
public interface HisMsgMapper {
    void initTable();

    void dropTable();

    long insert(@Param("type") String type, @Param("msg") String msg, @Param("role") long role);

    int update(@Param("id") long id, @Param("msg") String msg, @Param("role") long role);

    List<HisMsg> historyMsg(@Param("id") Long id, @Param("limit") int limit);

    List<HisMsg> listAll(@Param("offset") long offset, @Param("size") long size);

    Long count();
}

File: src/main/kotlin/com/github/bin/mapper/HisMsgMapper.kt
package com.github.bin.mapper

import com.github.bin.entity.msg.HisMsg
import org.apache.ibatis.annotations.Param

/**
 *  @Date:2023/4/9
 *  @author bin
 *  @version 1.0.0
 */
interface HisMsgMapper {
    fun initTable()
    fun dropTable()

    fun insert(@Param("type") type: String, @Param("msg") msg: String, @Param("role") role: Long): Long
    fun update(@Param("id") id: Long, @Param("msg") msg: String, @Param("role") role: Long): Int
    fun historyMsg(@Param("id") id: Long?, @Param("limit") limit: Int): List<HisMsg>
    fun listAll(@Param("offset") offset: Long, @Param("size") size: Long): List<HisMsg>
    fun count(): Long

}
